atomate/vasp/firetasks/electrode_tasks.py

```diff
@@ -0,0 +1,311 @@
+import math
+from collections import defaultdict
+
+from fireworks import FiretaskBase, explicit_serialize, FWAction, Firework, Workflow
+from pymatgen import Structure
+from pymatgen.analysis.structure_matcher import StructureMatcher
+
+from atomate.utils.utils import env_chk, get_logger
+from pymatgen.analysis.defects.utils import ChargeInsertionAnalyzer
+
+from atomate.vasp.config import DB_FILE
+from atomate.vasp.fireworks.core import OptimizeFW, StaticFW
+
+from atomate.vasp.database import VaspCalcDb
+from atomate.vasp.firetasks import pass_vasp_result
+from atomate.vasp.powerups import powerup_by_kwargs, POWERUP_NAMES
+
+__author__ = "Jimmy Shen"
+__email__ = "jmmshn@lbl.gov"
+
+logger = get_logger(__name__)
+
+SM_DICT = StructureMatcher().as_dict()
+
+
+def update_wf_keys(wf, fw_spec):
+    """
+    Helper function to allow specific keys to be propagated to newly created workflows
+    update the wf in place with the keys in the fw_spec dictionary
+    Args:
+        wf: workflow to modify
+        fw_spec: dictionary to be passed
+    """
+
+    PASS_KEYS = [
+        "working_ion",
+        "db_file",
+        "vasp_powerups",
+        "base_task_id",
+        "max_insertions",
+        "base_structure",
+        "vasptodb_kwargs",
+        "staticfw_kwargs",
+        "optimizefw_kwargs",
+        "structure_matcher",
+        "allow_fizzled_parents",
+        "volumetric_data_type",
+    ]
+    for k in PASS_KEYS:
+        if k in fw_spec:
+            for fw in wf.fws:
+                fw.spec.update({k: fw_spec.get(k, None)})
+
+
+@explicit_serialize
+class AnalyzeChgcar(FiretaskBase):
+    """
+    Process the charge density from a particular task.
+    fw_spec requires:
+    - "base_task_id" : the task_id for the charge density that is being analyzed
```

Comment:
Is there any reason you are passing these variables via the FW spec, rather than using `required_params` and `optional_params`?

If not, could you convert the Firetask to use those instead?

Reply:
So I looked at this one specifically, and even here since this relies on `task_fields_to_push` to work the `base_task_id` have to be stored in fw.spec.